dotCMS/src/main/java/com/dotmarketing/portlets/templates/business/TemplateAPIImpl.java

```diff
@@ -173,6 +185,70 @@ protected void save(WebAsset webAsset) throws DotDataException {
 		save((Template) webAsset);
 	}
 
+	@WrapInTransaction
+	public boolean publishTemplate(final Template template, final User user, final boolean respectFrontendRoles) {
+
+		Logger.debug(this, ()-> "Publishing the template: " + template.getIdentifier());
+		return Try.of(()->PublishFactory.publishAsset(template, user, respectFrontendRoles)).getOrElseThrow(e -> new RuntimeException(e));
+	}
+
+	@WrapInTransaction
+	public boolean unpublishTemplate(final Template template, final User user, final boolean respectFrontendRoles) {
+
+		Logger.debug(this, ()-> "Unpublishing the template: " + template.getIdentifier());
+		final Folder parent = Try.of(()->APILocator.getFolderAPI()
+				.findParentFolder(template, user, respectFrontendRoles)).getOrElseThrow(e -> new RuntimeException(e));
+		return Try.of(()->WebAssetFactory.unPublishAsset(template, user.getUserId(), parent))
+				.getOrElseThrow(e -> new RuntimeException(e));
+	}
+
+	@WrapInTransaction
+	public void unlock (final Template template, final User user) {
+
+		Try.run(()->this.versionableAPI.get().setLocked(template, false, user))
+				.getOrElseThrow(e -> new RuntimeException(e));
+	}
+
+	@WrapInTransaction
+	public boolean archive (final Template template, final User user, final boolean respectFrontendRoles) {
+
+		Logger.debug(this, ()-> "Doing archive of the template: " + template.getIdentifier());
+		if (Try.of(()->template.isLive()).getOrElseThrow(e -> new RuntimeException(e))) {
+
+			if (!this.unpublishTemplate(template, user, respectFrontendRoles)) {
```

Comment:
Codacy found an issue: [These nested if statements could be combined](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6042990)